kernel/kernel-api/src/main/java/io/delta/kernel/internal/TransactionMetadataFactory.java

```diff
@@ -246,6 +250,7 @@ private TransactionMetadataFactory(
       Optional<SnapshotImpl> latestSnapshotOpt,
       Optional<Metadata> initialNewMetadata,
       Optional<Protocol> initialNewProtocol,
+      Map<String, String> originalUserInputProperties,
```

Comment:
Can't we just use the table properties in `initialNewMetadata`?

Reply:
If there is no newMetadata -- no need to do this check anyways since there was no table properties input